hs_core/tasks.py

```diff
@@ -2,45 +2,49 @@
 
 from __future__ import absolute_import
 
+import logging
 import os
 import sys
 import traceback
 import zipfile
-import logging
-
-import requests
-
+from datetime import timedelta, date
 from xml.etree import ElementTree
 
-from rest_framework import status
-
+import requests
+from celery import shared_task
+from celery.schedules import crontab
+from celery.task import periodic_task
 from django.conf import settings
-from django.core.mail import send_mail
 from django.contrib.auth.models import User
+from django.core.mail import send_mail
+from rest_framework import status
 
-from celery.task import periodic_task
-from celery.schedules import crontab
-from celery import shared_task
-
-from hs_core.models import BaseResource
+from django_irods.icommands import SessionException
+from django_irods.storage import IrodsStorage
 from hs_core.hydroshare import utils
 from hs_core.hydroshare.hs_bagit import create_bag_files
 from hs_core.hydroshare.resource import get_activated_doi, get_resource_doi, \
     get_crossref_url, deposit_res_metadata_with_crossref
-from django_irods.storage import IrodsStorage
+from hs_core.models import BaseResource
 from theme.models import UserQuota, QuotaMessage
 from theme.utils import get_quota_message
 
-from django_irods.icommands import SessionException
-
-
 # Pass 'django' into getLogger instead of __name__
 # for celery tasks (as this seems to be the
 # only way to successfully log in code executed
 # by celery, despite our catch-all handler).
 logger = logging.getLogger('django')
 
 
+@periodic_task(ignore_result=True, run_every=crontab(minute=0, hour=0))
+def nightly_zips_cleanup():
+    # delete 2 days ago to avoid deleting zips just before midnight
+    date_folder = (date.today() - timedelta(2)).strftime('%Y-%m-%d')
+    irods_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+    zips_daily_date = "{irods}/zips/{daily_date}".format(irods=irods_prefix, daily_date=date_folder)

```

Comment:
@sblack-usu I'd suggest to use ```os.path.join()``` to form ```zips_daily_date``` path rather than a string concatenation.

Reply:
that's an irods path, not an os path.  os.path.join does validation on the paths

Reply:
@sblack-usu OK, if it is a absolute path, os.path.join will work for irods path as well. However, if you store temp zips always in the data zone, you don't need absolute path, but instead a relative path is sufficient and simpler. You can remove irods_prefix but instead just use path starting with zips/{daily_date}/...

Reply:
removed the irods prefix!  that cleans it up nicely.